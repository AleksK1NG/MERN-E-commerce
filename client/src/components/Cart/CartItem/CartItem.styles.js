import styled from 'styled-components'

export const CartItemContainer = styled.div`
  width: 100%;
  display: flex;
  height: 80px;
  margin-bottom: 15px;
`

CartItemContainer.displayName = 'CartItemContainer'

export const CartItemImage = styled.img`
  width: 30%;
`

CartItemImage.displayName = 'CartItemImage'

export const ItemDetailsContainer = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding: 10px 20px;
`
ItemDetailsContainer.displayName = 'ItemDetailsContainer'