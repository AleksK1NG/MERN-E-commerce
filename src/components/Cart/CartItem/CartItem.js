import React from 'react'
import { CartItemContainer, CartItemImage, ItemDetailsContainer } from './CartItem.styles'

const CartItem = ({ item: { imageUrl, price, name, quantity } }) => {
  return (
    <CartItemContainer>
      <CartItemImage src={imageUrl} alt="item" />
      <ItemDetailsContainer>
        <span>{name}</span>
        <span>
          {quantity} x ${price}
        </span>
      </ItemDetailsContainer>
    </CartItemContainer>
  )
}

export default React.memo(CartItem)
