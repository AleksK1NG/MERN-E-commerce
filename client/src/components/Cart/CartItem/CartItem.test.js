import React from 'react'
import { shallow } from 'enzyme'
import { CartItem } from './CartItem'

describe('test CartItem component', () => {
  const mockItem = {
    imageUrl: '',
    price: 1,
    name: 'hats',
    quantity: 1
  }

  const wrapper = shallow(<CartItem item={mockItem} />)

  it('should render CartItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have span with name', () => {
    expect(
      wrapper
        .find('span')
        .first()
        .text()
    ).toEqual(mockItem.name)
  })
})
