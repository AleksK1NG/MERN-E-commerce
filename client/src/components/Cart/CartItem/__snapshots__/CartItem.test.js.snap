// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CartItem component should render CartItem component 1`] = `ShallowWrapper {}`;
